rfc/src/rfcs/0005-should-fail-attr.md

```diff
@@ -0,0 +1,197 @@
+
+### Alternative #1: Name
+
+Our first choice is made in the name of the attribute.
+It should be noted that Rust has a similarly named attribute for unit tests: [the `#[should_panic]` attribute](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics).
+At first, one may think that `#[kani::should_fail]` and `#[should_panic]` have the same semantics.
+However, if we keep thinking about this, we'll realize that:
+ 1. A panic in Rust should always result in a failure in Kani, but
+ 2. A failure in Kani doesn't always result in a panic in Rust.
+
+Our running example shows an instance of (2): an overflow error doesn't result in a panic in Rust.
+So it's better to avoid `panic` in the name.
+
+We have also considered two alternatives for the expectation: `should` and `expect`.
+To be honest, we avoid `expect` altogether for two reasons:
+ - We may consider a more granular approach to indicate expectations regarding individual checks and cover statements in the future. One possible name for the attribute is `#[kani::expect]`.
+ - We heavily use this word for testing in Kani: there is an `expected` mode, which works with `*.expected` files. Other modes also use such files.
+
+### Alternative #2: Granularity
+
+As mentioned earlier, users may want to express more granular expectations for their harnesses.
+For example, a user may want to specify that a given check results in a failure.
+
+Again, there is a relation between this and the [the `#[should_panic]` attribute](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#testing-panics) for Rust tests.
+More specifically, the `#[should_panic]` attribute may receive an argument `expected` which allows users to specify the expected panic string:
+
+```rust
+    #[test]
+    #[should_panic(expected = "Divide result is zero")]
+    fn test_specific_panic() {
+        divide_non_zero_result(1, 10);
+    }
+```
+
+In principle, it wouldn't be a problem to extend this proposal to include the `expected` argument.
+The implementation could compare the `expected` string against property descriptions.
+
+However, there could be some problems later with this approach:
+ * What if users want to specify multiple `expected` strings?
+ * What if users don't want to only check for failures?
+ * In the previous case, would they expect the overall verification to fail or not?
+
+We don't know if these will be requirements in the future.
+If they were, then we'd need to come up with language to state expectations about properties (checks or cover statements).
+A good option to consider here is adding the single-string `expected` argument (as in `#[should_panic]`) and see how it's used.
+It could be the cause of a breaking change in the future, but it would provide us with useful data in the meatime.
+
+## Open questions
+
+The main question I'd like to get resolved are:
+ * Do we want to extend `#[kani::should_fail]` with an `expected` field?
```

Comment:
I'm worried that `should_fail` is too open since it includes any UB.

Reply:
This was answered in my long comment.